libs/StateSync.Managed/src/NetworkManager.cs

```diff
@@ -0,0 +1,222 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.MixedReality.Sharing.StateSync
+{
+    public interface NetworkManager
+    {
+        NetworkConnection GetConnection(InternedBlobRef connectionString);
+
+        bool PollMessage(NetworkListener listener);
+    }
+
+    /// <summary>
+    /// Maintains the cache of C++ wrappers for the created channels.
+    /// </summary>
+    public class NetworkManagerPInvokeWrapper
```

Comment:
nit: make internal, move to a different file.

Reply:
Done.